mesonbuild/compilers/compilers.py

```diff
@@ -872,7 +872,7 @@ def has_multi_link_arguments(self, args, env):
 
     def get_cross_extra_flags(self, environment, link):
         extra_flags = []
-        if self.is_cross and environment:
+        if environment.is_cross_build() and self.for_machine >= MachineChoice.HOST:
```

Comment:
I can't tell just by looking what this comparison is supposed to do. It needs a cleaner interface.